src/app_service/service/transaction/dtoV2.nim

```diff
@@ -0,0 +1,79 @@
+# import json, json_serialization, stint
+
+# import ../network/dto, ../token/dto
+
+# include  app_service/common/json_utils
+
+import json, strutils, stint, json_serialization
+
+import
+  web3/ethtypes
+
+include  ../../common/json_utils
+import ../network/dto, ../token/dto
+
+type
+  SuggestedLevelsForMaxFeesPerGasDto* = ref object
+    low*: UInt256
+    medium*: UInt256
+    high*: UInt256
+
+type
+  TransactionPathDtoV2* = ref object
+    processorName*: string
+    fromChain*: NetworkDto
+    toChain*: NetworkDto
+    fromToken*: TokenDto
+    amountIn*: UInt256
+    amountInLocked*: bool
+    amountOut*: UInt256
+    suggestedLevelsForMaxFeesPerGas*: SuggestedLevelsForMaxFeesPerGasDto
+    txBaseFee*: UInt256
+    txPriorityFee*: UInt256
+    txGasAmount*: uint64
+    txBonderFees*: UInt256
+    txTokenFees*: UInt256
+    txL1Fee*: UInt256
+    approvalRequired*: bool
+    approvalAmountRequired*: UInt256
+    approvalContractAddress*: string
+    approvalBaseFee*: UInt256
+    approvalPriorityFee*: UInt256
+    approvalGasAmount*: uint64
+    approvalL1Fee*: UInt256
+    estimatedTime*: int
+
+proc toSuggestedLevelsForMaxFeesPerGasDto*(jsonObj: JsonNode): SuggestedLevelsForMaxFeesPerGasDto =
+  result = SuggestedLevelsForMaxFeesPerGasDto()
+  var value: int
+  if jsonObj.getProp("low", value):
+    result.low = stint.fromHex(UInt256, $value)
+  if jsonObj.getProp("medium", value):
+    result.medium = stint.fromHex(UInt256, $value)
+  if jsonObj.getProp("high", value):
+    result.high = stint.fromHex(UInt256, $value)
+
+proc toTransactionPathDtoV2*(jsonObj: JsonNode): TransactionPathDtoV2 =
+  result = TransactionPathDtoV2()
+  discard jsonObj.getProp("ProcessorName", result.processorName)
```

Comment:
hmm first time I see usage of the default marshalling in status-go (https://github.com/status-im/status-go/pull/5426/files#diff-15991723d09f0a991be1caafa148d1840acd316437041021a7ac2ee392541da5L73), I think we normally set up the field names explicitly (so, for example, changing a variable name in the status-go struct doesn't break API compatibility). I'm not sure if it's something we enforce though.